### PR TITLE
Setup redux

### PR DESCRIPTION
**Pull request by allie-rae**

### CODE REVIEW

**Review by zimashima — APPROVED** [review]
I was there when it happened

**Review by danimal-johnson — APPROVED** [review]
Good teamwork! Enjoyed the livestream. :)

File: src/redux/dog/reducer.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+interface DogState {
+  name: string | null;
+  sex: string;
+  age: number | null;
+  isVaccinated: boolean;
+  breed: string | null;
+  breedId: number | null;
+  isFixed: boolean;
+  bio: string | null;
+  pic: string | null;
```
Comment (zimashima) [inline src/redux/dog/reducer.tsx:12]: for future reference, can you alphabetize this? :)

File: src/redux/dog/reducer.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+const initialState: DogState = {
+  name: null,
+  sex: 'female',
+  age: null,
+  isVaccinated: false,
+  breed: null,
+  breedId: null,
+  isFixed: false,
+  bio: null,
+  pic: null,
```
Comment (zimashima) [inline src/redux/dog/reducer.tsx:37]: this, too :)

File: src/redux/dog/reducer.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+const dogCrud = (state: DogState, action: Action) => {
+  state.name = action.payload.name;
+  state.sex = action.payload.sex;
+  state.age = action.payload.age;
+  state.isVaccinated = action.payload.is_vaccinated;
+  state.breed = action.payload.breed;
+  state.isFixed = action.payload.is_fixed;
+  state.bio = action.payload.bio;
+  state.pic = action.payload.pic;
```
Comment (zimashima) [inline src/redux/dog/reducer.tsx:50]: Also this :)

File: src/redux/dog/reducer.tsx (hunk 2 above)
Comment (zimashima) [reply src/redux/dog/reducer.tsx:37]:
```suggestion
```

File: src/redux/dog/reducer.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+const createDog = createAction('CREATE/DOG');
+const updateDog = createAction('UPDATE/DOG');
+const getDog = createAction('GET/DOG');
+const deleteDog = createAction('DELETE/DOG');
+
+const initialState: DogState = {
+  // temperament: null,
+  // size: null,
+};
```
Comment (danimal-johnson) [inline src/redux/dog/reducer.tsx:29]: const initialState: DogState = {
  // temperament: null,
  // size: null,
};
```suggestion
const initialState: DogState = {
  name: null,
  sex: 'female',
  age: null,
  isVaccinated: false,
  breed: null,
  breedId: null,
  isFixed: false,
  bio: null,
  pic: null,
  // temperament: null,
  // size: null,
};
```

File: src/redux/dog/reducer.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+interface Action {
+  type: string;
+  payload: any;
+};
+
+const createDog = createAction('CREATE/DOG');
+const updateDog = createAction('UPDATE/DOG');
+const getDog = createAction('GET/DOG');
```
Comment (zimashima) [inline src/redux/dog/reducer.tsx:25]:
```suggestion
const getDog = createAction('GET/DOG');

```

**Pull request merged**